### ISSUE TITLE
Missing summaryWithMetadata google-cloud-discoveryengine

### ISSUE THREAD

**Issue by tpcgold** [googleapis/google-cloud-python#12358]
The summaryWithMetadata [Documentation](https://cloud.google.com/generative-ai-app-builder/docs/get-search-summaries?hl=en#get-citations) is not available on the google-cloud-discoveryengine library.

This makes VertexAI Search & Conversation pretty unusable via Python library.

Example (Note: don't use discoveryengine_v1 (as ExtractiveContentSpec is missing there).
Basically an updated version of this [documentation](https://cloud.google.com/generative-ai-app-builder/docs/preview-search-results?hl=en#structured-unstructured):
```python
from typing import List

from google.api_core.client_options import ClientOptions
from google.cloud import discoveryengine

# TODO(developer): Uncomment these variables before running the sample.
# project_id = "YOUR_PROJECT_ID"
# location = "YOUR_LOCATION"                    # Values: "global", "us", "eu"
# data_store_id = "YOUR_DATA_STORE_ID"
# search_query = "YOUR_SEARCH_QUERY"

def search(
    project_id: str,
    location: str,
    data_store_id: str,
    search_query: str,
    filter: str = None,
) -> List[discoveryengine.SearchResponse]:
    #  For more information, refer to:
    # https://cloud.google.com/generative-ai-app-builder/docs/locations#specify_a_multi-region_for_your_data_store
    client_options = (
        ClientOptions(api_endpoint=f"{location}-discoveryengine.googleapis.com")
        if location != "global"
        else None
    )

    # Create a client
    client = discoveryengine.SearchServiceClient(client_options=client_options)

    # The full resource name of the search engine serving config
    # e.g. projects/{project_id}/locations/{location}/dataStores/{data_store_id}/servingConfigs/{serving_config_id}
    serving_config = client.serving_config_path(
        project=project_id,
        location=location,
        data_store=data_store_id,
        serving_config="default_config",
    )

    # Optional: Configuration options for search
    # Refer to the `ContentSearchSpec` reference for all supported fields:
    # https://cloud.google.com/python/docs/reference/discoveryengine/latest/google.cloud.discoveryengine_v1.types.SearchRequest.ContentSearchSpec
    content_search_spec = discoveryengine.SearchRequest.ContentSearchSpec(
        # For information about snippets, refer to:
        # https://cloud.google.com/generative-ai-app-builder/docs/snippets
        snippet_spec=discoveryengine.SearchRequest.ContentSearchSpec.SnippetSpec(
            return_snippet=True
        ),
        # For information about search summaries, refer to:
        # https://cloud.google.com/generative-ai-app-builder/docs/get-search-summaries
        summary_spec=discoveryengine.SearchRequest.ContentSearchSpec.SummarySpec(
            summary_result_count=5,
            include_citations=True,
            ignore_adversarial_query=True,
            ignore_non_summary_seeking_query=True,
        ),
        extractive_content_spec=discoveryengine.SearchRequest.ContentSearchSpec.ExtractiveContentSpec(
            max_extractive_answer_count=1
        ),
    )

    # Refer to the `SearchRequest` reference for all supported fields:
    # https://cloud.google.com/python/docs/reference/discoveryengine/latest/google.cloud.discoveryengine_v1.types.SearchRequest
    request = discoveryengine.SearchRequest(
        serving_config=serving_config,
        query=search_query,
        page_size=10,
        content_search_spec=content_search_spec,
        filter=filter,
        query_expansion_spec=discoveryengine.SearchRequest.QueryExpansionSpec(
            condition=discoveryengine.SearchRequest.QueryExpansionSpec.Condition.AUTO,
        ),
        spell_correction_spec=discoveryengine.SearchRequest.SpellCorrectionSpec(
            mode=discoveryengine.SearchRequest.SpellCorrectionSpec.Mode.AUTO
        ),
    )

    response = client.search(request)
    print(response)

    return response

**Comment by parthea** [comment]
Hi @tpcgold ,

Thanks for reporting this issue. 

The `summary_with_metadata` field was added to the client library in https://github.com/googleapis/google-cloud-python/pull/12293 and is available as of `google-cloud-discoveryengine==0.11.8`

https://github.com/googleapis/google-cloud-python/releases/tag/google-cloud-discoveryengine-v0.11.8

https://github.com/googleapis/google-cloud-python/blob/24f085185f0c90cbb71fb68a3bdf45fac1a5fe5a/packages/google-cloud-discoveryengine/google/cloud/discoveryengine_v1/types/search_service.py#L1032-L1033

I'm going to transfer this issue to the `python-docs-samples` repository so that the code sample [here](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/HEAD/discoveryengine/search_sample.py) can be updated.

**Comment by holtskinner** [comment]
No code changes are needed. Once the client library was updated, the code should work as expected.
